src/JakobSteinn/HTMLFormValidator/RuleProviders/AbstractRuleProvider.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace JakobSteinn\HTMLFormValidator\RuleProviders;
+
+abstract class AbstractRuleProvider implements RuleProviderContract
```

Comment:
This is actually exactly what `Zend\Validator` does (and the symfony validators, and respect validation, and others): I would suggest not re-designing that, since the validation lib is quite solid there.

Actually, you may just leverage [`Zend\InputFilter\Factory`](http://framework.zend.com/manual/current/en/modules/zend.input-filter.intro.html) to initialize an array of validators (and just use this lib to extract the array data to pass to that factory)